tests/bluetooth/bsim_bt/bsim_test_audio/src/vcs_client_test.c

```diff
@@ -0,0 +1,730 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BT_VCS_CLIENT
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio/vcs.h>
+
+#include "common.h"
+
+
+#define VOCS_DESC_SIZE 64
+#define AICS_DESC_SIZE 64
+
+static struct bt_conn_cb conn_callbacks;
+extern enum bst_result_t bst_result;
+
+static struct bt_vcs vcs;
+static volatile bool g_bt_init;
+static volatile bool g_is_connected;
+static volatile bool g_mtu_exchanged;
+static volatile bool g_discovery_complete;
+static volatile bool g_write_complete;
+
+static volatile uint8_t g_volume;
+static volatile uint8_t g_mute;
+static volatile uint8_t g_flags;
+static volatile int16_t g_vocs_offset;
+static volatile uint32_t g_vocs_location;
+static char g_vocs_desc[VOCS_DESC_SIZE];
+static volatile int8_t g_aics_gain;
+static volatile uint8_t g_aics_input_mute;
+static volatile uint8_t g_aics_mode;
+static volatile uint8_t g_aics_input_type;
+static volatile uint8_t g_aics_units;
+static volatile uint8_t g_aics_gain_max;
+static volatile uint8_t g_aics_gain_min;
+static volatile bool g_aics_active = 1;
+static char g_aics_desc[AICS_DESC_SIZE];
+static volatile bool g_cb;
+static struct bt_conn *g_conn;
+
+static void vcs_state_cb(struct bt_conn *conn, int err, uint8_t volume,
+			 uint8_t mute)
+{
+	if (err) {
+		FAIL("VCS state cb err (%d)", err);
+		return;
+	}
+
+	g_volume = volume;
+	g_mute = mute;
+
+	g_cb = true;
+}
+
+static void vcs_flags_cb(struct bt_conn *conn, int err, uint8_t flags)
+{
+	if (err) {
+		FAIL("VCS flags cb err (%d)", err);
+		return;
+	}
+
+	g_flags = flags;
+
+	g_cb = true;
+}
+
+static void vocs_state_cb(struct bt_conn *conn, struct bt_vocs *inst, int err,
+			  int16_t offset)
+{
+	if (err) {
+		FAIL("VOCS state cb err (%d)", err);
+		return;
+	}
+
+	g_vocs_offset = offset;
+
+	g_cb = true;
+}
+
+static void vocs_location_cb(struct bt_conn *conn, struct bt_vocs *inst,
+			     int err, uint32_t location)
+{
+	if (err) {
+		FAIL("VOCS location cb err (%d)", err);
+		return;
+	}
+
+	g_vocs_location = location;
+
+	g_cb = true;
+}
+
+static void vocs_description_cb(struct bt_conn *conn, struct bt_vocs *inst,
+				int err, char *description)
+{
+	if (err) {
+		FAIL("VOCS description cb err (%d)", err);
+		return;
+	}
+
+	if (strlen(description) > sizeof(g_vocs_desc) - 1) {
+		printk("Warning: VOCS description (%zu) is larger than buffer (%zu)\n",
+		       strlen(description), sizeof(g_vocs_desc) - 1);
+	}
+
+	strncpy(g_vocs_desc, description, sizeof(g_vocs_desc) - 1);
+
```

Comment:
Set terminating zero?

Reply:
`strncpy` should handle that for us.

Reply:
"The  strncpy()  function  is  similar,  except that at most n bytes of src are copied.  Warning: If there is no null byte among the first n bytes of src, the string placed in dest will not be null-terminated."
(This is from the Linux manpage, but I expect zephyr is the same.)

The code checks against "sizeof() - 1", and also copies that many bytes.  I expected the last byte (the -1) to be used for the termination (in case of truncation), but it isn't.